segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookkeeperLogFactoryTests.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.bookkeeper;
+
+import io.pravega.common.Timer;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import org.apache.bookkeeper.client.api.BookKeeper;
+import org.apache.bookkeeper.test.BookKeeperClusterTestCase;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Test to exercise the re-creation of the Bookkeeper client upon successive log creation attempts.
+ */
+public class BookkeeperLogFactoryTests extends BookKeeperClusterTestCase {
```

Comment:
Why do you subclass from this? 

Can't you do it like we test `BookKeeperLog`? We should have everything we need there.

Reply:
Moved the test to `BookkeeperLog`.